KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/CuePointsManager.java

```diff
@@ -44,16 +31,15 @@
 	private final static int DEFAULT_SYNC_POINTS_INTERVAL_IN_MS = 8000;
 	private final static String KALTURA_SYNC_POINTS_INTERVAL_PROPERTY = "KalturaSyncPointsInterval";
 	private static final long START_SYNC_POINTS_DELAY = 0;
+	private final ConcurrentHashMap<IMediaStream,CuePointManagerLiveStreamListener> streams;
```

Comment:
Why do you manage the listeners list? Once the steam destroyed all references to the listener will be destroyed and the GC will clean the listener.